iwf/workflow_state.py

```diff
@@ -0,0 +1,108 @@
+from abc import ABC, abstractmethod
+from typing import TypeVar, Generic
+
+from iwf.command_request import CommandRequest
+from iwf.command_results import CommandResults
+from iwf.communication import Communication
+from iwf.persistence import Persistence
+from iwf.state_decision import StateDecision
+from iwf.workflow_context import WorkflowContext
+from iwf.workflow_state_options import WorkflowStateOptions
+
+T = TypeVar("T")
+
+
+class WorkflowState(ABC, Generic[T]):
+    """WorkflowState is the interface to define a workflow state."""
+
+    @abstractmethod
+    def get_input_type(self) -> type[T]:
```

Comment:
this does need to be provided by a subclass? https://stackoverflow.com/questions/48572831/how-to-access-the-type-arguments-of-typing-generic

Reply:
type is "needed" but T is not. Type will be needed for deserailize the payload into the state input. I can be None if no input. Or some primitive types

Reply:
Great idea! Removed this method:D